resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/CustomUserStorageProvider.java

```diff
@@ -0,0 +1,315 @@
+        log.debug("getUserByEmail({})", email);
+        try (Connection c = getConnection(this.model)) {
+            PreparedStatement st = this.prepareUserQuery(c, String.format("WHERE email = '%s'", email), "");
+            log.debug("getUserByEmail: st={}", st);
+            st.execute();
+            ResultSet rs = st.getResultSet();
+            if (rs.next()) {
+                return new UserAdapter(ksession, realm, model, UserObject.fromJoinedResultSet(rs));
+            } else {
+                return null;
+            }
+        } catch (SQLException ex) {
+            throw new RuntimeStorageException(ex);
+        }
+    }
+
+    
+    // UserQueryProvider implementation
+    @Override
+    public int getUsersCount(RealmModel realm) {
+        log.debug("getUsersCount: realm={}", realm.getName());
+        try (Connection c = getConnection(this.model)) {
+            Statement st = c.createStatement();
+            st.execute("select count(*) from public.users");
+            log.debug("getUsersCount: st={}", st);
+            ResultSet rs = st.getResultSet();
+            rs.next();
+            return rs.getInt(1);
+        } catch (SQLException ex) {
+            throw new RuntimeStorageException(ex);
+        }
+    }
+
+    @Override
+    public Stream<UserModel> getGroupMembersStream(RealmModel realm, GroupModel group, Integer firstResult,
+            Integer maxResults) {
+        log.debug("getGroupMembersStream: realm={}", realm.getName());
+
+        try (Connection c = getConnection(this.model)) {
+            PreparedStatement st = this.prepareUserQuery(c, "",
+                    String.format("order by email limit %o offset %o", maxResults, firstResult));
+            log.debug("getGroupMembersStream: st={}", st);
+            st.execute();
+            ResultSet rs = st.getResultSet();
+            List<UserModel> users = new ArrayList<>();
+            while (rs.next()) {
+                users.add(new UserAdapter(ksession, realm, model, UserObject.fromJoinedResultSet(rs)));
+            }
+            return users.stream();
+        } catch (SQLException ex) {
+            throw new RuntimeStorageException(ex);
+        }
+    }
+
+    @Override
+    public Stream<UserModel> searchForUserStream(RealmModel realm, Map<String, String> params, Integer firstResult,
+            Integer maxResults) {
+        return getGroupMembersStream(realm, null, firstResult, maxResults);
+    }
+
+    @Override
+    public Stream<UserModel> searchForUserByUserAttributeStream(RealmModel realm, String attrName, String attrValue) {
+        log.debug("searchForUserByUserAttributeStream: realm={}, attrName={}, attrValue={}", realm.getName(),
+                attrName, attrValue);
+        try (Connection c = getConnection(this.model)) {
+            PreparedStatement st = this.prepareUserQuery(c,
+                    String.format("WHERE %s = %s", attrName, attrValue),
+                    String.format("order by email"));
```

Comment:
suggestion(non-blocking): since there are no format arguments, we can remove the call to `String.format`
```suggestion
                    "order by email");
```

Reply:
done